src/windows/process.rs

```diff
@@ -500,26 +587,33 @@ fn get_cmd_line(handle: HANDLE) -> Vec<String> {
         }
         buffer_copy.push(0);
 
-        // Get argc and argv from command line
-        let mut argc = MaybeUninit::<i32>::uninit();
-        let argv_p =
-            winapi::um::shellapi::CommandLineToArgvW(buffer_copy.as_ptr(), argc.as_mut_ptr());
-        if argv_p.is_null() {
-            return Vec::new();
-        }
-        let argc = argc.assume_init();
-        let argv = std::slice::from_raw_parts(argv_p, argc as usize);
-
-        let mut res = Vec::new();
-        for arg in argv {
-            let len = libc::wcslen(*arg);
-            let str_slice = std::slice::from_raw_parts(*arg, len);
-            res.push(String::from_utf16_lossy(str_slice));
-        }
+        get_cmdline_from_buffer(buffer_copy.as_ptr())
+    }
+}
+
+#[allow(clippy::cast_ptr_alignment)]
+fn get_cmd_line_new(handle: HANDLE) -> Vec<String> {
+    unsafe {
+        if let Some(buffer) = 
+            ph_query_process_variable_size(handle, ProcessCommandLineInformation) {
 
-        winapi::um::winbase::LocalFree(argv_p as *mut _);
+            let buffer = (*(buffer.as_ptr() as *const UNICODE_STRING)).Buffer;
+
+            get_cmdline_from_buffer(buffer)
+        } else {
+            vec![]
+        }
+    }
+}
 
-        res
+fn get_cmd_line(handle: HANDLE) -> Vec<String> {
+    if *WINDOWS_8_1_OR_NEWER
+    {
```

Comment:
This is a very weird formatting. :)

Reply:
I don't know why, but on this project cargo fmt doesn't work on my computer <_<

Reply:
Dark magic. :D